File: support_tools/swigwin-3.0.12/Examples/test-suite/python/disown_runme.py
from disown import *

a = A()

tmp = a.thisown

a.thisown = 0
if a.thisown:
    raise RuntimeError

a.thisown = 1
if (not a.thisown):
    raise RuntimeError

a.thisown = tmp
if (a.thisown != tmp):
    raise RuntimeError


b = B()

b.acquire(a)

if a.thisown:
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/disown_runme.js
var disown = require("disown");

var a = new disown.A();
var tmp = a.thisown;
a.thisown = 0
if (a.thisown) {
  throw new Error("Failed.");
}
a.thisown = 1
if (!a.thisown) {
  throw new Error("Failed.");
}
a.thisown = tmp
if (a.thisown != tmp) {
  throw new Error("Failed.");
}

var b = new disown.B();
b.acquire(a);
if (a.thisown) {
  throw new Error("Failed.");
}
